File: solutions/python/grains/1/grains.py
def square(number):
    if number < 1 or number > 64:
        raise ValueError("square must be between 1 and 64")
    return 2 ** (number-1)

def total():
    result = 0
    for i in range(1, 65):
        result += square(i)

    return result

File: solutions/c/grains/1/grains.c
#include "grains.h"

uint64_t square(uint8_t index)
{
    if (index == 1) return 1;
    uint64_t result;
    result = square(index - 1) * 2;
    return result;
}
uint64_t total(void)
{
    uint64_t sum = 0;
    int i;
    for (i = 0; i <= 64; i++)
    {
        sum += square(i);
    }
    return sum;
}